sentry-spring/src/test/kotlin/io/sentry/spring/SentrySpringIntegrationTest.kt

```diff
@@ -0,0 +1,148 @@
+package io.sentry.spring
+
+import com.nhaarman.mockitokotlin2.check
+import com.nhaarman.mockitokotlin2.verify
+import io.sentry.core.IHub
+import io.sentry.core.Sentry
+import io.sentry.core.SentryEvent
+import io.sentry.core.SentryOptions
+import io.sentry.core.transport.ITransport
+import java.lang.RuntimeException
+import org.assertj.core.api.Assertions.assertThat
+import org.awaitility.kotlin.await
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.boot.web.server.LocalServerPort
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.core.userdetails.User
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.security.core.userdetails.UserDetailsService
+import org.springframework.security.crypto.factory.PasswordEncoderFactories
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.security.provisioning.InMemoryUserDetailsManager
+import org.springframework.security.web.authentication.AnonymousAuthenticationFilter
+import org.springframework.test.context.junit4.SpringRunner
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RunWith(SpringRunner::class)
+@SpringBootTest(
+    classes = [App::class],
+    webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT,
+    properties = ["sentry.dsn=http://key@localhost/proj", "sentry.send-default-pii=true"]
+)
+class SentrySpringIntegrationTest {
```

Comment:
This test class is almost a duplicate from Sentry Spring Boot integration. I am not sure yet if/how we can reduce the duplication keeping the same confidence level.

Reply:
should we maybe add a comment to this class and the "duplicated" one about each other? so when touching either this one or the other one, we know that they need to keep kinda synced